web/src/flatpickr.js

```diff
@@ -31,15 +31,9 @@ export function show_flatpickr(element, callback, default_timestamp, options = {
         disableMobile: true,
         time_24hr: user_settings.twenty_four_hour_time,
         onKeyDown(selectedDates, dateStr, instance, event) {
-            if (is_numeric_key(event.key)) {
-                // Don't attempt to get_keydown_hotkey for numeric inputs
-                // as it would return undefined.
-                return;
-            }
-
             const hotkey = get_keydown_hotkey(event);
 
-            if (["tab", "shift_tab"].includes(hotkey.name)) {
+            if (hotkey && ["tab", "shift_tab"].includes(hotkey.name)) {
```

Comment:
Is the correct behavior to stop propagation for keys not on this list? We might want to do an early return as we did before.

I tracked down bea41e975dbd8771698b32809692c95e820479bb with `git log -S`, which introduced the code you're removing here, but haven't investigated what that means for this question.

Reply:
I took a closer look at the commit that you linked to.  The code is very different from what we have today.  In the original code we were deliberately returning early WITHOUT stop propagation for numeric keys and for "backspace" and "delete."

I don't know why the comments were changed from

```javascript
// Don't stopPropagation for numeric inputs, let them be handled normally
```

and

```javascript
// Don't stopPropagation for backspace or delete, let them be handled normally
```

to

```javascript
  // Don't attempt to get_keydown_hotkey for numeric inputs
  // as it would return undefined.
```

or why we are no longer stopping propagation for "backspace" and "delete."

I think the answer is to return early, then, for numeric keys, the "backspace" key and the "delete" key, and also to use an `if...else` block to avoid the redundant stop propagation call.

If this sounds correct, I am happy to push the changes.

Reply:
One can probably figure out with `git log -Sbackspace` on that file, but I suspect for backspace/delete, the story is that we determined the `flatpickr` native behavior for those keys was reasonable (possibly it was a workaround for an upstream bug to not do it there in the first place? This old comment is poor in not explaining why it's doing that).

I think returning early sounds correct; the effect of `stopPropagation` is going to be disabling whatever upstream `flatpickr` keyboard code exists, and we presumably only want to do that when we've got an explicit reason to do so.

Reply:
Thanks for sharing `git log -S` with me.  It's hard to follow git history with git blame when code is moved to a new file, as it was here.

I'll try to comb through the history and see if I can glean any more context.  Then I will push up a revision and note any new context discovered.

Reply:
So I did a little digging with `git log -S` (thank you for this!!!) and found [e17c96d](https://github.com/zulip/zulip/commit/e17c96d86dee11df8425567260236985a2313cc2).  This commit explicitly changes the comment to the following:

> // Don't attempt to get_keydown_hotkey for numeric inputs
> // as it would return undefined.

And the commit description states:

> In the previous commit (https://github.com/zulip/zulip/commit/bea41e975dbd8771698b32809692c95e820479bb) we
> introduced a bug by using `hotkey` instead of `hotkey.name`, further
> debugging revealed that the conditional was unnecessary and as such
> has been removed in this commit, **the comment for the is_numeric**
> **conditional has also been changed to explain its actual purpose.**

I believe that my current code/commit is handling the original intent correctly.

I did some manual testing with my current code/branch.  I can use `backspace`, `tab`, `shift` + `tab`, numeric keys and letters without any errors or bugs.

I also tried commenting out the stop propagation on line 53.  All of the above-mentioned keys still worked properly and without throwing errors.

Is there anything I should test outside of the date/time picker that could be affected?

Do we actually need the stop propagation code on line 53?

Reply:
We can definitely leave in the stopPropagation logic.

1) Should I go ahead and add an `else` condition to avoid calling stopPropagation twice for hot keys?

2) Would you like me to take the comments explaining the purpose of the `onKeyDown` event handler from #23356 and add them to this PR?

Reply:
Yeah, both of those changes sound reasonable to me.

Reply:
@timabbott -- Changes made.